test/heap-types.wast

```diff
@@ -169,7 +170,9 @@
     )
     (drop
       (block $out (result (ref $struct.B))
-        (drop
+        ;; set the value to a local with type $struct.A, showing that the value
+        ;; flowing out has the right type
+        (local.set $temp.A
```

Comment:
Sorry what I meant was not value flowing out but the target block's return type...

Reply:
Sorry I didn't know what I was saying is [already tested](https://github.com/WebAssembly/binaryen/pull/3451#discussion_r545553499) in the validator. Nevermind.